client/src/types/user.ts

```diff
@@ -2,12 +2,13 @@ import { ModelMetadata } from './model';
 
 export type Credentials = {
   email: string;
-  secret?: string;
+  secret: string;
 }
 
 export type UserData = Credentials & {
```

Comment:
I think `secret` is still going to be a required property on the `UserData` type since it's required on `Credentials`.

`string & (string | undefined)` &rarr; `(string & string) | (string & undefined)` &rarr; `string | never` &rarr; `string` :(

I hate that it doesn't just override it like you might think but I bet there's some good explanation somewhere with something to do with sets or something or the other 👎 

We can do
```suggestion
export type UserData = Omit<Credentials, 'secret'> & {
```